src/cmake/dependencies.cmake

```diff
@@ -102,7 +102,8 @@ endif()
 set(RECASTNAVIGATION_DEMO OFF CACHE BOOL "RECASTNAVIGATION_DEMO" FORCE)
 set(RECASTNAVIGATION_TESTS OFF CACHE BOOL "RECASTNAVIGATION_TESTS" FORCE)
 set(RECASTNAVIGATION_EXAMPLES OFF CACHE BOOL "RECASTNAVIGATION_EXAMPLES" FORCE)
-set(RECASTNAVIGATION_STATIC ON CACHE BOOL "RECASTNAVIGATION_STATIC" FORCE)
+set(BUILD_SHARED_LIBS OFF CACHE BOOL "BUILD_SHARED_LIBS" FORCE)
```

Comment:
Recast now uses the more standard BUILD_SHARE_LIBS CMake config to determine whether to do a static build.

Now also requires the GNUInstallDirs to be set.

Reply:
Just a note -- on older CMake, Bullet *has to be* built as shared because the static subproject build is broken beyond repair: https://github.com/facebookresearch/habitat-sim/blob/e821de1fbb4198353a1f7f1f1db940942fa5ba6c/src/cmake/dependencies.cmake#L164-L178 

Which points to a problem with the global `BUILD_SHARED_LIBS` -- there's no way to set it differently for different subprojects, which is why Magnum has its own variable as well. Is it possible to control this with some Recast-specific variable? Otherwise we force it to `OFF` in one place and to `ON` in another and it's ... a mess :)

Reply:
@mosra @aclegg3 Unfortunately, no the RECAST specific variable was removed in an update. Can we store it to a temp variable and reset it somewhere? This already is an issue for the ASSIMP submodule anyway so it's not like it's really introducing a new problem. Thoughts?

Reply:
@mosra Relevant PR: https://github.com/recastnavigation/recastnavigation/pull/439